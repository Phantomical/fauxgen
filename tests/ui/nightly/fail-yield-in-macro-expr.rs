#![feature(generators, generator_trait)]
#![deny(deprecated)]

macro_rules! delay {
    ($value:expr) => { () };
}

#[fauxgen::generator]
fn gen() {
    let _ = delay!(yield);
}

fn main() {}
