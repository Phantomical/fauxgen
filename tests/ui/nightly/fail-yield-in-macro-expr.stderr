error: use of deprecated function `_::warning`: `yield` expressions in macros are not expanded by fauxgen
  --> tests/ui/nightly/fail-yield-in-macro-expr.rs:10:20
   |
10 |     let _ = delay!(yield);
   |                    ^^^^^
   |
note: the lint level is defined here
  --> tests/ui/nightly/fail-yield-in-macro-expr.rs:2:9
   |
2  | #![deny(deprecated)]
   |         ^^^^^^^^^^
